### PR TITLE
refactor!(crypto): Move some caches out of the CryptoStore implementations

### PR DESCRIPTION
**Pull request by poljar**
This moves the caches we have for tracked users out of the Sled, and IndexedDB, CryptoStore into the Store struct. The Store struct is a wrapper for the CryptoStore trait, so this is the natural place where these caches should live.

### CODE REVIEW

File: crates/matrix-sdk-crypto/src/store/mod.rs [hunk 1]
```diff
@@ -587,6 +614,99 @@ impl Store {
...
+        self.load_tracked_users().await?;
+
+        Ok(self.users_for_key_query_cache.iter().map(|u| u.clone()).collect())
+    }
+
+    /// See the docs for [`crate::OlmMachine::tracked_users()`].
+    pub async fn tracked_users(&self) -> Result<HashSet<OwnedUserId>> {
+        self.load_tracked_users().await?;
+
+        Ok(self.tracked_users_cache.to_owned().iter().map(|u| u.clone()).collect())
```
Comment (jplatte) [inline crates/matrix-sdk-crypto/src/store/mod.rs:708]: Why `.to_owned()` before `.iter()` here? Seems unnecessary.

File: crates/matrix-sdk-indexeddb/src/crypto_store.rs [hunk 2]
```diff
@@ -756,38 +750,24 @@ impl IndexeddbCryptoStore {
...
-                false => JsValue::FALSE,
-            },
-        )?;
+        for (user, dirty) in users {
+            os.put_key_val(
+                &JsValue::from_str(user.as_str()),
+                &match dirty {
+                    true => JsValue::TRUE,
+                    false => JsValue::FALSE,
+                },
```
Comment (jplatte) [inline crates/matrix-sdk-indexeddb/src/crypto_store.rs:765]:
```suggestion
                &dirty.into(),
```

Comment (poljar) [reply crates/matrix-sdk-indexeddb/src/crypto_store.rs:765]: This doesn't work directly as we have a `&bool`. Will use `JsValue::from()` instead.

File: crates/matrix-sdk-crypto/src/store/mod.rs (hunk 1 above)
Comment (poljar) [reply crates/matrix-sdk-crypto/src/store/mod.rs:708]: Indeed, thanks.

File: crates/matrix-sdk-crypto/src/store/mod.rs [hunk 3]
```diff
@@ -40,7 +40,6 @@
 
 pub mod caches;
 mod memorystore;
-
 #[cfg(any(test, feature = "testing"))]
```
Comment (jplatte) [inline crates/matrix-sdk-crypto/src/store/mod.rs:43]:
```suggestion

#[cfg(any(test, feature = "testing"))]
```

**Review by jplatte — APPROVED** [review]

**Pull request merged**